### ISSUE TITLE
Inventaire with GunDB as the data store and sync instead of CouchDB

### ISSUE THREAD

**Issue by trosel** [inventaire/inventaire#297]
I read that it is a goal to move away from centralization. So this could be part of that step at some point.

[Gun](https://github.com/amark/gun) is a decentralized datastore that follows the same synchronization method as Couch.

Gun could handle User authentication as well as the library data.

I believe DTube and Archive.org are both using it today.

https://github.com/amark/gun/wiki/Awesome-GUN

**Comment by maxlath** [comment]
Indeed, while the bibliographic should, in my opinion, be centralized, users inventory could be decentralized (see [discussion on ActivityPub](https://github.com/inventaire/inventaire/issues/187)).

What I struggle to get is how access control would be handled in such a P2P way: do you know if it is possible to finely control which piece of data is visible by who with Gun?

Other interesting candidates for P2P inventory data sharing:
* [Scuttlebot](https://scuttlebot.io)

**Comment by trosel** [comment]
>What I struggle to get is how access control would be handled in such a P2P way: do you know if it is possible to finely control which piece of data is visible by who with Gun?

See this [FAQ answer](https://gun.eco/docs/FAQ#does-gun-have-some-form-of-acl-access-rights-a-name-acl-a-)

Also, please come chat on the [gitter](https://gitter.im/amark/gun ). Much smarter people than I are there. I'm only a hobbyist.

**Comment by trosel** [comment]
Revisiting this topic because I have renewed interest in a decentralized goodreads haha.

> What I struggle to get is how access control would be handled in such a P2P way: do you know if it is possible to finely control which piece of data is visible by who with Gun?

This is indeed tricky. I think both Gun and Scuttlebutt assume a public propogation to the ends of the earth. But at least with Gun's SEA library, it's easy to create new keys on the fly. So there could be some logic that shares your library only with "friends", for example.

Also, something to keep in mind is this social library 

- https://docs.iris.to/#socialnetwork
- https://github.com/irislib/iris-lib

CC @amark
